src/backend_connection/authenticated_cmds.rs

```diff
@@ -0,0 +1,67 @@
+use std::sync::Arc;
+
+use pyo3::{
+    pyclass, pyclass_init::PyObjectInit, pymethods, PyClassInitializer, PyErr, PyObject, PyResult,
+    PyTypeInfo, Python,
+};
+
+use libparsec::{client_connection, protocol::authenticated_cmds};
+
+use crate::{
+    addrs::BackendOrganizationAddr,
+    api_crypto::SigningKey,
+    ids::DeviceID,
+    protocol::{AuthenticatedPingRep, AuthenticatedPingRepOk, AuthenticatedPingRepUnknownStatus},
+    runtime::FutureIntoCoroutine,
+};
+
+use super::CommandErrorExc;
+
+#[pyclass]
+pub(crate) struct AuthenticatedCmds(Arc<client_connection::AuthenticatedCmds>);
+
+#[pymethods]
+impl AuthenticatedCmds {
+    #[new]
+    fn new(
+        server_url: BackendOrganizationAddr,
+        device_id: DeviceID,
+        signing_key: SigningKey,
+    ) -> PyResult<Self> {
+        let auth_cmds =
+            client_connection::client::generate_client(signing_key.0, device_id.0, server_url.0);
+        Ok(Self(Arc::new(auth_cmds)))
+    }
+
+    fn ping(&self, ping: String) -> FutureIntoCoroutine {
+        let auth_cmds = self.0.clone();
+
+        FutureIntoCoroutine::from(async move {
+            auth_cmds
+                .ping(ping)
+                .await
+                .map(AuthenticatedPingRep)
```

Comment:
Why do you map if your next step is to get `.0` ?

Reply:
I forgot to remove it. Done.